design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,169 @@
 
-**Note: This step does not cover the scenario where a user has a unique table, in which case they need to create an additional table to add a relationship.**
+### Scenario 1b: Manually from the 'Table Constraints' settings
 
-#### 2. The user selects a table from the list of available tables
-- The user selects the table on the 'one side of the 'one-to-many' relationship.
+#### Steps for 1b
 
-#### 3. The user selects a column belonging to the selected table to create the link
-- The primary key column is pre-selected by default
-#### The user sets the new column as a foreign key for a column in the 'artist' table
-- If the column had values, this action would clear those unless they exactly matched contents from the linked column
-#### The user retrieves records from the linked table and adds them to the new column
-- Autocomplete should allow users to complete this action without additional clicks
-- When no results are available, we should show an option to add a new record
-#### 4. The user removes the foreign key constraint from the column
-- The system will assign the column the type of the column that was linked (if the linked column was a duration type, then it should be assigned as such)
+- A user wants to set up a foreign key constraint to single or multiple columns to a currently active table.
+- The user opens the table options by clicking on the table name label in the toolbar area. From the menu, the user selects the option 'Table Constraints'
+- This opens a dialog from which the user can set all supported constraint types.
+- The user sees a layout with two panels. On one side, all the existing constraints are listed. On the other side, a form with the constraints configuration is displayed. The form corresponds to the currently active list item.
+- The list panel contains actions at the bottom of the panel. From there, the user can choose to add or delete an item from the list. The user clicks on 'Add' to create a new constraint.
+- The user is presented with a form. From there, they select the type of constraint to be added. The user selects the 'Foreign Key' constraint option from the list.
+- The user selects single or multiple columns they wish to apply the constraint to.
+- The user selects a table to be referenced by these columns.
+- The user selects a column in the reference table from which the column will match values. By default, this is set to the 'Primary Key' column in the referenced table. When changing this field, the user can select only columns with a UNIQUE or PRIMARY KEY constraint.
 
----
+## Scenario 2: User Edits an Existing Foreign Key Constraint
+
+### Scenario 2a: From the column header menu
+
+#### Steps for 2a
+
+- The user hovers over the foreign key reference link below the column name.
+- Information about the constraints applied to the column is displayed in a tooltip.
+- The user clicks on the foreign key reference link below the column name.
```

Comment:
In #786 @pavish changed the column header to trigger the column header menu when clicking anywhere inside the header (instead of clicking only on the arrow), which I like. With this spec, we'd move from having one interaction inside the column header to having three: the column menu, the constraint hover, and the constraint edit. I think that's too many. Plus, I don't like the idea of having separate click/hover actions on the same screen area because once we begin to support touch devices we'll lose the hover functionality. 

I'd prefer to move the functionality from the tooltip to be inside the column header menu. Clicking anywhere in the column header would bring up the column header menu. When the column has an FK constraint, there would be a section in that menu to explain it. The user could click somewhere within that section to bring up the constraints modal and edit the constraint.

Reply:
@ghislaineguerin I like showing the referenced table on the header. But I agree with @seancolsen that we should not open the constraints modal by clicking on it.
* Editing constraints is a rarely performed operation. But opening the column dropdown is essential for anything users would want to do with the column. So, it makes sense to only use the entire column header space to open the dropdown.
* As Sean described, we can show it within the dropdown, have a section for it and open the constraints modal when user clicks on it.

Reply:
I agree, I have removed the additional click target from the column header.